File: c/cz/mg/vulkan/PFN_vkCreatePipelineCache.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkCreatePipelineCache_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pPipelineCache, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreatePipelineCache f = (PFN_vkCreatePipelineCache)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkPipelineCacheCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkPipelineCache*)jniLongToPointer(pPipelineCache))
    );
}

File: cz/mg/vulkan/PFNvkCreatePipelineCache.java
package cz.mg.vulkan;

public class PFNvkCreatePipelineCache extends VkFunctionPointer {
    public PFNvkCreatePipelineCache() {
    }

    public PFNvkCreatePipelineCache(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCreatePipelineCache(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkCreatePipelineCache(long value) {
        setValue(value);
    }

    public PFNvkCreatePipelineCache(VkInstance instance) {
        super(instance, new VkString("vkCreatePipelineCache"));
    }

    public void call(VkDevice device, VkPipelineCacheCreateInfo pCreateInfo, VkAllocationCallbacks pAllocator, VkPipelineCache pPipelineCache, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pPipelineCache != null ? pPipelineCache.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getNullAddress());
    }

    private static native void call(long vkaddress, long device, long pCreateInfo, long pAllocator, long pPipelineCache, long rval);
}
